submarine-commons/cluster-server/src/test/java/org/apache/submarine/cluster/ClusterMultiNodeTest.java

```diff
@@ -0,0 +1,186 @@
+
+    // mock cluster manager server
+    String cluster[] = clusterAddrList.split(",");
+    try {
+      for (int i = 0; i < 3; i++) {
+        String[] parts = cluster[i].split(":");
+        String clusterHost = parts[0];
+        int clusterPort = Integer.valueOf(parts[1]);
+
+        Class clazz = ClusterManagerServer.class;
+        Constructor constructor = clazz.getDeclaredConstructor();
+        constructor.setAccessible(true);
+        ClusterManagerServer clusterServer = (ClusterManagerServer) constructor.newInstance();
+        clusterServer.initTestCluster(clusterAddrList, clusterHost, clusterPort);
+
+        clusterServers.add(clusterServer);
+      }
+    } catch (Exception e) {
+      LOGGER.error(e.getMessage(), e);
+    }
+
+    for (ClusterManagerServer clusterServer : clusterServers) {
+      clusterServer.start();
+    }
+
+    // mock cluster manager client
+    try {
+      Class clazz = ClusterManagerClient.class;
+      Constructor constructor = null;
+      constructor = clazz.getDeclaredConstructor();
+      constructor.setAccessible(true);
+      clusterClient = (ClusterManagerClient) constructor.newInstance();
+      clusterClient.start(metaKey);
+    } catch (NoSuchMethodException | InstantiationException
+        | IllegalAccessException | InvocationTargetException e) {
+      LOGGER.error(e.getMessage(), e);
+    }
+
+    // Waiting for cluster startup
+    boolean clusterIsStartup = false;
+    int wait = 0;
+    while (wait++ < 100) {
+      if (clusterIsStartup() && clusterClient.raftInitialized()) {
+        LOGGER.info("ClusterMultiNodeTest::wait {}(ms) found cluster leader", wait * 3000);
+        clusterIsStartup = true;
+        break;
+      }
+      try {
+        Thread.sleep(3000);
+      } catch (InterruptedException e) {
+        LOGGER.error(e.getMessage(), e);
+      }
+    }
+
+    assertEquals(clusterIsStartup, true);
+
+    Thread.sleep(5000);
+    assertEquals(true, clusterIsStartup());
+    LOGGER.info("ClusterMultiNodeTest::startCluster <<<");
+  }
+
+  @AfterClass
+  public static void stopCluster() {
+    LOGGER.info("ClusterMultiNodeTest::stopCluster >>>");
+    if (null != clusterClient) {
+      clusterClient.shutdown();
+    }
+    for (ClusterManagerServer clusterServer : clusterServers) {
+      clusterServer.shutdown();
+    }
+    LOGGER.info("ClusterMultiNodeTest::stopCluster <<<");
+  }
+
+  static boolean clusterIsStartup() {
+    boolean foundLeader = false;
+    for (ClusterManagerServer clusterServer : clusterServers) {
+      if (!clusterServer.raftInitialized()) {
+        LOGGER.warn("clusterServer not Initialized!");
+        return false;
+      }
+      if (clusterServer.isClusterLeader()) {
+        foundLeader = true;
+      }
+    }
+
+    if (!foundLeader) {
+      LOGGER.warn("Can not found leader!");
+      return false;
+    }
+
+    LOGGER.info("cluster startup!");
+    return true;
+  }
+
+  @Test
+  public void getClusterServerMeta() {
```

Comment:
All test method should named with `test` prefix.

Reply:
Done.